pommerman/network/ionc/__init__.py

```diff
@@ -0,0 +1,135 @@
+    """Get the intent of the user"""
+    i = ui.ask_choice(Strings.intent.value, [
+        Strings.intent_match.value, Strings.intent_room.value,
+        Strings.intent_replay.value, Strings.intent_exit.value
+    ])
+    if i == Strings.intent_match.value:
+        match(network)
+    elif i == Strings.intent_room.value:
+        match(network, True)
+    elif i == Strings.intent_replay.value:
+        replay(network)
+    elif i == Strings.intent_exit.value:
+        exit(0)
+
+
+def match(network, room=False):
+    """Facilitates playing of matches"""
+    from os import getcwd
+    from sys import path
+    from importlib import import_module
+    if room:
+        room = str(ui.ask_string(Strings.room_code.value))
+    path.append(getcwd())
+    agent = import_module(ui.ask_string(Strings.match_import.value))
+    agent_class = ui.ask_string(Strings.match_class_name.value)
+    if agent_class not in agent.__dir__():
+        ui.fatal("Cannot find class in module file")
+    agent = getattr(agent, agent_class)
+    if getattr(agent, "act"):
+        ui.info(ui.green, Strings.match_agent_success.value)
+    agent = agent()
+    ui.info(ui.yellow, Strings.server_comm.value)
+    network.join_list(room)
+    ui.info(Strings.match_variant.value, ui.yellow, network.mode)
+    ui.info(ui.yellow, Strings.match_wait.value)
+    network.waitm()
+    ui.info(Strings.match_run.value, "#" + network.match_id)
+    while (True):
+        match_obj = network.match_get()
+        if match_obj[0] is not False:
+            action = agent.act(match_obj[0], Discrete(6))
+            action_response = network.send_move(action, match_obj[1])
+            if action_response[0] is False:
+                ui.fatal(Strings[action_response[1]].value)
+        elif not match_obj[0] and isinstance(match_obj[1], int):
+            agent.episode_end(match_obj[1])
+            if match_obj[1] == 1:
+                ui.info(Strings.match_won.value)
+            if match_obj[1] == -1:
+                ui.info(Strings.match_loss_draw.value)
+            break
+        else:
+            ui.info(ui.yellow, Strings[match_obj[1]].value)
+    ui.info(Strings.match_replay.value, ui.yellow, network.match_id)
+    if ui.ask_yes_no(Strings.match_ask_replay.value):
+        replay(network.match_id)
+    else:
+        intent()
+
+
+def replay(network, id=False):
+    """Retrieve and play replays"""
+    if not id:
+        id = str(ui.ask_string(Strings.replay_prompt.value))
+        if id is None:
+            ui.fatal(Strings.e_invalid_id.value)
+    ui.info(Strings.server_replay_p1.value, ui.yellow, "#" + str(id), ui.reset,
+            Strings.server_replay_p2.value)
+    r = network.get_replay(id)
+    if not r[0]:
+        ui.fatal(Strings[r[1]].value)
+    ui.info(Strings.replay_start.value, ui.yellow, "#" + str(id))
+    r = r[1]
+    from pommerman import make
+    from pommerman.agents import BaseAgent
+    from time import sleep
+    from numpy import array
```

Comment:
Why not put these at top?

Reply:
I'm very conservative with my imports and only do so if it's required so I put them here instead at the top as the importing is conditional then.